plugins/extend-give-wp/src/support/load-assets.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Load the assets
+ *
+ * @package     spiralWebDB\ExtendGiveWP
+ * @since       1.0.0
+ * @author      Robert A. Gadon
+ * @link        https://spiralwebdb.com
+ * @license     GNU General Public License 2.0+
+ */
+
+namespace spiralWebDB\ExtendGiveWP;
+
+add_action( 'give_pre_form', __NAMESPACE__ . '\render_form_featured_image_and_caption' );
+/*
+ * Render donation form featured image and caption.
+ *
+ * @since 1.0.0
+ * @param int $_id  Post ID of the donation form.
+ *
+ * @return void
+ */
+function render_form_featured_image_and_caption( $_id = 1349 ) {
+	// Get the post object for the attachment via get_post().
+	$post    = get_post( 1411 );
```

Comment:
Why is the post ID hard-coded here? Why is this problematic? There are multiple reasons:

- The post ID on your machine _may no_t be the same post ID on the live production website.
- The post ID could change over time.

A better approach is to provide a way to get the post for donation form by configuration:

- site admin selection, ie the site admin can select which one it is via Customizer or a settings screen
- post metadata that's configurable with the editor for that specific post/page/cpt

The first option is likely better. Why? The 2nd option applies a metabox/attribute to every single post type. Using that mindset, the first option is more targeted without impacting or cluttering the editor for every similar post type.

Reply:
@hellofromtonya I'm in agreement with you that hard-coding in the `$_id` argument is not good practice. I chose to do this so I could render _something_ to the specific donation form called by the Give plugin. I'm also not keen on passing in a specific post ID to the `get_post()` method. Again, I needed to specify the ID for the post object in order to render content to the donation form. 

As I look at the documentation for WP (Theme and Plugin handbooks, respectively), I see that the Customizer is recommended for configuring themes, and admin menu and submenu pages are recommended for selecting or inputting plugin options. Since these inputs would target custom callback arguments for the plugin, it seems that an options submenu is the proper route to pursue. 

Let's discuss requirements for that feature.

Reply:
@hellofromtonya See commits 
[06a8989](https://github.com/rgadon107/cornerstone/pull/52/commits/06a8989fee487e4c18d4a0a0dd75101f91ec6387) and 
[8dc43c7](https://github.com/rgadon107/cornerstone/pull/52/commits/8dc43c774c31a25b8f05f38d1881abf17003e06e). 

Both the callback and view were updated. They still don't resolve the issue of passing in specific IDs for donation form and attachment. But they do reduce the number of instances where those values are declared.